perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -440,8 +460,11 @@ public void removeGroupFromResource(PerunSession sess, Group group, Resource res
 			throw new InternalErrorException("Group " + group + " and resource " + resource + " belongs to the different VOs");
 		}
 
-		// Check if the group was defined on the resource
-		if (!this.getAssignedGroups(sess, resource).contains(group)) {
+		// Check if the group was defined on the resource (with any GroupResource status)
+		List<Group> groupsOnResource = getGroupAssignments(sess, resource, List.of()).stream()
+			.map(g -> g.getEnrichedGroup().getGroup())
+			.collect(Collectors.toList());
+		if (!groupsOnResource.contains(group)) {
```

Comment:
I don't think we need to change this method, it will be replaced with a new one.